include/rocksdb/io_status.h

```diff
@@ -266,4 +266,8 @@ inline IOStatus status_to_io_status(Status&& status) {
   }
 }
 
+inline Status io_status_to_status(IOStatus& io_s) {
```

Comment:
@anand1976 Are you ok this this change?

Reply:
Seems ok to me. Can we make it ```io_statuc_to_status(IOStatus&& io_s)```? I think doing ```std::move``` on a lvalue reference is a bit of an anti-pattern.

Reply:
This seems unnecessary (and violating Google C++ style on names). What's wrong with

    Status(io_s)

or

    static_cast<const Status&>(io_s);

? If you're going to motivate use of move semantics, I wouldn't use a case where it's eligible for copy elision. https://en.cppreference.com/w/cpp/language/copy_elision